cache/cache_reservation_manager.h

```diff
@@ -90,6 +118,7 @@ class CacheReservationManager {
   std::shared_ptr<Cache> cache_;
   bool delayed_decrease_;
   std::atomic<std::size_t> cache_allocated_size_;
+  std::atomic<std::size_t> memory_used_;
```

Comment:
My feeling is memory_order_relaxed is an ambiguous name because it could mean (a) no special memory ordering, or (b) especially loose memory ordering. Since in my understanding it actually means (a), I think of it the same as updating a non-atomic like a plain `int` in a mutex, with the additional guarantee that out-of-mutex observers cannot see partial writes (because it's `atomic`).

Reply:
Good point 👍